### ISSUE TITLE
[Feature Request]: Rspack should validate tsconfig.json

### ISSUE THREAD

**Issue by magic-akari** [web-infra-dev/rspack#2489]
### What problem does this feature solve?

I noticed that rspack using swc instead of directly using tsc for transpilation.
Although there are some restrictions [mentioned in the documentation](https://www.rspack.dev/guide/language-support.html#typescript), I found that there were no notifications provided to me when the `isolatedModules` option was not enabled in tsconfig.json.
A warning and optional fix tips would be more user-friendly for users.

Additionally, swc lists more issues that require attention. Please refer to https://swc.rs/docs/migrating-from-tsc .

### What does the proposed API of configuration look like?

There are several optional approaches:

1. Use independent sub-commands, such as `rspack doctor`.
2. Enable checking by adding a boolean flag in `rspack.config.js`.

**Comment by hardfist** [comment]
good suggestions! I think checking is better than another subcommand

**Comment by dhruvkelawala** [comment]
I can have a look into this

**Comment by hyf0** [comment]
I think this is important for users and DX. To tell/warn users what the appropriate options users should have in `tsconfig.json`.

A friendly ping for @dhruvkelawala. Do you still have an interest in this?

**Comment by dhruvkelawala** [comment]
@hyf0 Yes, I do. I will get on it asap! Sorry have been busy with work.

**Comment by hyf0** [comment]
> @hyf0 Yes, I do. I will get on it asap! Sorry have been busy with work.

Take your time. This is not an urgent feature.

**Comment by dhruvkelawala** [comment]
I see @suxin2017 has already created a PR. So I will unassign this issue to me

**Comment by hyf0** [comment]
Sync up by Rspack team:

Considering users may use `ts-loader` in the future or other tools that care about `tsconfig.json`, it's kind of not proper for Rspack to validate `tsconfig.json`.

I would close this issue and label it with `wontfix`. But welcome if anyone has suggestions on it.
